src/django_components/component_registry.py

```diff
@@ -1,25 +1,16 @@
-from typing import TYPE_CHECKING, Callable, Dict, List, NamedTuple, Optional, Set, Type, TypeVar
+from typing import TYPE_CHECKING, Callable, Dict, NamedTuple, Optional, Set, Type, TypeVar
 
 from django.template import Library
 
+from django_components.library import Tags, is_tag_protected, mark_protected_tags, register_tags_from_formatter
+from django_components.tag_formatter import get_tag_formatter
+
 if TYPE_CHECKING:
     from django_components.component import Component
 
 _TComp = TypeVar("_TComp", bound=Type["Component"])
 
 
-PROTECTED_TAGS = [
```

Comment:
Moved the logic that's concerned with working with Django's Library object into `library.py`. Also, added a test that checks that the tag protection works correctly.

https://github.com/EmilStenstrom/django-components/pull/572/files#diff-0abc741bfcebe9debcf5eb70d01e66db1ccb9036877f195680d25bf73e9e9083R144